ProcessOptimizer/space/space.py

```diff
@@ -1,917 +1,940 @@
+        return Space(input)
+
+
+def check_dimension(dimension, transform=None):
+    """Turn a provided dimension description into a dimension object.
+
+    Checks that the provided dimension falls into one of the
+    supported types. For a list of supported types, look at
+    the documentation of ``dimension`` below.
+
+    If ``dimension`` is already a ``Dimension`` instance, return it.
+
+    Parameters
+    ----------
+    * `dimension`:
+        Search space Dimension.
+        Each search dimension can be defined either as
+
+        - a `(lower_bound, upper_bound)` tuple (for `Real` or `Integer`
+          dimensions),
+        - a `(lower_bound, upper_bound, "prior")` tuple (for `Real`
+          dimensions),
+        - as a list of categories (for `Categorical` dimensions), or
+        - an instance of a `Dimension` object (`Real`, `Integer` or
+          `Categorical`).
+
+    * `transform` ["identity", "normalize", "onehot" optional]:
+        - For `Categorical` dimensions, the following transformations are
+          supported.
+
+          - "onehot" (default) one-hot transformation of the original space.
+          - "identity" same as the original space.
+
+        - For `Real` and `Integer` dimensions, the following transformations
+          are supported.
+
+          - "identity", (default) the transformed space is the same as the
+            original space.
+          - "normalize", the transformed space is scaled to be between 0 and 1.
+
+    Returns
+    -------
+    * `dimension`:
+        Dimension instance.
+    """
+    if isinstance(dimension, Dimension):
+        return dimension
+
+    if not isinstance(dimension, (list, tuple, np.ndarray)):
+        raise ValueError("Dimension has to be a list or tuple.")
+
+    # A `Dimension` described by a single value is assumed to be
+    # a `Categorical` dimension. This can be used in `BayesSearchCV`
+    # to define subspaces that fix one value, e.g. to choose the
+    # model type, see "sklearn-gridsearchcv-replacement.ipynb"
+    # for examples.
+    if len(dimension) == 1:
+        return Categorical(dimension, transform=transform)
+
+    if len(dimension) == 2:
+        if any(
+            [isinstance(d, (str, bool)) or isinstance(d, np.bool_) for d in dimension]
+        ):
+            return Categorical(dimension, transform=transform)
+        elif all([isinstance(dim, numbers.Integral) for dim in dimension]):
+            return Integer(*dimension, transform=transform)
+        elif any([isinstance(dim, numbers.Real) for dim in dimension]):
+            return Real(*dimension, transform=transform)
+        else:
+            raise ValueError(
+                "Invalid dimension {}. Read the documentation for"
+                " supported types.".format(dimension)
+            )
+
+    if len(dimension) == 3:
+        if any([isinstance(dim, (float, int)) for dim in dimension[:2]]) and dimension[
+            2
+        ] in ["uniform", "log-uniform"]:
```

Comment:
```suggestion
        if any([isinstance(dim, (float, int)) for dim in dimension[:2]]) and 
            dimension[2] in ["uniform", "log-uniform"]:
```
The old formatting reads better.